inst/private/assert_have_python_and_sympy.m

```diff
@@ -16,35 +16,126 @@
 %% License along with this software; see the file COPYING.
 %% If not, see <http://www.gnu.org/licenses/>.
 
-function assert_have_python_and_sympy(pyexec)
-%private function
+%% -*- texinfo -*-
+%% @defun  assert_have_python_and_sympy (pyexec)
+%% @defunx assert_have_python_and_sympy (pyexec, verbose)
+%% Check that Python and SymPy are installed and working.
+%%
+%% @seealso{sympref}
+%% @end defun
+
+function assert_have_python_and_sympy (pyexec, verbose)
+
+  minsympyver = '1.0';
 
-  [st, out] = system([pyexec ' -c "print(\"hello world\")"']);
-  if ( (st ~= 0) || (~ strcmp(strtrim(out), 'hello world')) )
-    error('OctSymPy:nopython', ...
-          ['Cannot run the Python executable "%s"\n' ...
-           '    Python and SymPy are needed for most Symbolic features.\n' ...
-           '    Is Python installed?  Is your "path" configured correctly?'], ...
-          pyexec)
+  if (nargin < 2)
+    verbose = false;
   end
 
+  if (verbose)
+    disp ('')
+    disp ('Symbolic package diagnostics')
+    disp ('============================')
+    disp ('')
+    disp ('Python and SymPy are needed for most features of the Symbolic package.')
+    disp ('')
+    fprintf ('The Python interpreter is currently: "%s".\n', pyexec)
+    disp ('')
+    disp ('Computers may have more than one Python interpreter installed.  Depending')
+    disp ('on how your system is configured, you may need to select a different one.')
+    disp ('See "help sympref" for details, but for example, to use Python 3, try')
```

Comment:
I will try to rephrase this: sounds like something is wrong and action is required.